staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/compatibility.go

```diff
@@ -422,15 +428,28 @@ func (c *CompatibilityTestOptions) runPreviousVersionTest(t *testing.T, gvk sche
 	}
 	protoAfterRoundTrip := protoBytes.Bytes()
 
+	jsonNeedsRemove := false
+	yamlNeedsRemove := false
+	protoNeedsRemove := false
+
 	expectedJSONAfterRoundTrip, expectedYAMLAfterRoundTrip, expectedProtoAfterRoundTrip, _ := read(previousVersionDir, gvk, ".after_roundtrip", usedFiles)
 	if len(expectedJSONAfterRoundTrip) == 0 {
 		expectedJSONAfterRoundTrip = jsonBeforeRoundTrip
+	} else if bytes.Equal(jsonBeforeRoundTrip, expectedJSONAfterRoundTrip) {
+		t.Errorf("JSON after_roundtrip file is identical and should be removed")
```

Comment:
How did we get such a file in the first place?

What if one or both buffers is not in canonical form?

Reply:
> How did we get such a file in the first place?

probably iterating on a PR in a way that `make update` created the after_roundtrip (like starting a new optional field as a non-pointer, `make update` created an after_roundtrip file, then later fixing the new field to be a pointer and `make update` regenerates `after_roundtrip` (since it exists) but the content is now identical to the original fixture